kernel/thread.c

```diff
@@ -39,6 +39,8 @@ LOG_MODULE_DECLARE(os, CONFIG_KERNEL_LOG_LEVEL);
 static struct k_spinlock z_thread_monitor_lock;
 #endif /* CONFIG_THREAD_MONITOR */
 
+static struct k_spinlock z_obj_lock;
```

Comment:
This seems kinda needless to me.  Surely the code will end up smaller if you just check the field for NULL, no?